idaes/apps/grid_integration/bidder.py

```diff
@@ -0,0 +1,774 @@
+    options, e.g. time.
+
+    Arguments:
+        df: the dataframe we are interested in
+        bus: the bus ID we want [int]/ the bus name we want [str]
+        date: the date we want [str]
+        hour: the hour we want [int]
+        generator: the generator ID [str]
+        fuel_type: generator fuel, e.g. Coal [str]
+    Returns:
+        df: a dataframe that has the information we specified.
+    '''
+
+    # get data given bus id
+    if bus is not None:
+        # in the original rts-gmlc dataset there is a Bus ID col
+        if 'Bus ID' in df.columns:
+            df = df.loc[(df['Bus ID'] == bus)]
+
+        # in the prescient result data we have to extract bus id from other col
+        # e.g. gennerator name col
+        elif 'Generator' in df.columns:
+            # convert the type to str
+            bus = str(bus)
+            # find the rows that starts with the bus name
+            searchrows = df['Generator'].str.startswith(bus)
+            df = df.loc[searchrows,:]
+
+        elif 'Bus' in df.columns:
+            df = df.loc[(df['Bus'] == bus)]
+
+    # get data given date
+    if date is not None:
+        df = df.loc[(df['Date'] == date)]
+
+    # get data given hour
+    if hour is not None:
+        df = df.loc[(df['Hour'] == hour)]
+
+    # get data given hour
+    if generator is not None:
+
+        # Similarly this is for Prescient result data
+        if 'Generator' in df.columns:
+            df = df.loc[(df['Generator'] == generator)]
+        # this is for rts-gmlc dataset
+        elif 'GEN UID' in df.columns:
+            df = df.loc[(df['GEN UID'] == generator)]
+
+    # get data given fuel
+    if fuel_type is not None:
+        df = df.loc[df['Fuel'] == fuel_type]
+
+    return df
+
+class DAM_thermal_bidding:
+
+    n_scenario = 10
+
+    def __init__(self,rts_gmlc_data_dir, price_forecast_file,horizon = 48,generators = None):
+
+        '''
+        Initializes the class object by doing building the bidding model.
+
+        Arguments:
+            rts_gmlc_data_dir: the RTS-GMLC data directory.
+            price_forecast_file: the directory we saved the price forecast for bidding
+            horizon: the length of the planning horizon of the model.
+            generators: a list of generators in RTS-GMLC
+
+        Returns:
+            None
+        '''
+
+        # read the price forecasts
+        self.price_forecasts = pd.read_csv(price_forecast_file,index_col = ['Date','Hour'])
+
+        self.model_data = self.assemble_model_data(generator_names = generators, \
+                                                   rts_gmlc_data_dir = rts_gmlc_data_dir)
+        self.model = self.build_thermal_bidding_model(plan_horizon = horizon,
+                                                      segment_number = 4)
+        self.bids_result_list = []
+
+    @staticmethod
+    def assemble_model_data(generator_names, rts_gmlc_data_dir, **kwargs):
```

Comment:
Does it make sense for generate_names and any RTS-GMLC data to live in self? Then this would not be a static method.